src/snowflake/snowpark/ml/transformer.py

```diff
@@ -0,0 +1,499 @@
+#
+# Copyright (c) 2012-2022 Snowflake Computing Inc. All rights reserved.
+#
+
+
+from typing import List, Optional, Tuple, Union
+
+from snowflake.snowpark._internal.utils import (
+    TempObjectType,
+    random_name_for_temp_object,
+)
+from snowflake.snowpark.dataframe import DataFrame
+from snowflake.snowpark.functions import (
+    array_agg,
+    array_cat,
+    array_construct,
+    array_construct_compact,
+    array_insert,
+    array_slice,
+    col,
+    iff,
+    lit,
+)
+from snowflake.snowpark.ml.utils import encoder_fit, scaler_fit
+
+
+class Transformer:
+    def __init__(
+        self,
+        input_cols: Optional[List[str]] = None,
+        output_cols: Optional[List[str]] = None,
+    ) -> None:
+        self._input_cols = input_cols
+        self._output_cols = output_cols
+        self._fitted = False
+
+    @property
+    def input_cols(self) -> List[str]:
+        return self._input_cols
+
+    @input_cols.setter
+    def input_cols(self, value: List[str]) -> None:
+        self._input_cols = value
+
+    @property
+    def output_cols(self) -> List[str]:
+        return self._output_cols
+
+    @output_cols.setter
+    def output_cols(self, value: List[str]) -> None:
+        self._output_cols = value
+
+    def fit(self, df: DataFrame) -> "Transformer":
```

Comment:
```suggestion
    def fit(self, df: DataFrame) -> None:
```
In this base class, we return nothing so the type hint should be fixed. The same for transform().